src/com/insightdatascience/xlnet_learn2learn/app.py

```diff
@@ -0,0 +1,58 @@
+#coding: utf8
+#author: Hairong Wang
+
+from flask import Flask,request,render_template
+import logging
+from logging import Formatter, FileHandler
+import optparse
+import json
+import os
+import re
+import collections
+import copy
+import json
+import time
+import datetime
+
+#debug_submodel = None
+
+app = Flask(__name__)
+
+def setup_app(app):
+  logging.basicConfig(filename='app.log',level=logging.INFO)
+
+setup_app(app)
+
+@app.route("/")
+def html_display():
+  return render_template("index.html")
+
+@app.route("/query", methods=['POST'])
+def get_query_and_context():
+  start_time = time.time()
```

Comment:
Could you add docstrings in appropriate places? They don't have to be long though. Here is an example of the Docstring conventions: https://www.python.org/dev/peps/pep-0257/